### PR TITLE
Initial stab at a snapcraft.yaml file

### PR DESCRIPTION
**Pull request by jwrober**
Needs to be back ported to stable to work.

See #1674

### CODE REVIEW

File: dist/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      - fonts-linuxlibertine
+      - libdw1
+      - liblua5.3-0
+      - libsdl2-2.0-0
+      - libsdl2-mixer-2.0-0
+      - libsqlite3-0
+      - libunwind8
+      - python3
+      - libreadline8
+      - zlib1g
```
Comment (lmoureaux) [inline dist/snapcraft.yaml:84]: Some of those are not needed:
```suggestion
      - liblua5.3-0
      - libsdl2-2.0-0
      - libsdl2-mixer-2.0-0
```

**Review by lmoureaux — COMMENTED** [review]
Got the following warning:
```
The 'snap' directory is meant specifically for snapcraft, but it contains the following non-snapcraft-related paths, which is unsupported and will cause unexpected behavior:
- freeciv21-client.png

If you must store these files within the 'snap' directory, move them to 'snap/local', which is ignored by snapcraft.
```

Testing done:

* Build the snap (takes forever)
* Install it (very long compared to normal snaps)
* Open the modpack installer
  * :warning: It seems to have the same icon as the game
* Install RoundSquare
* Close the modpack installer
* Open Freeciv21
* Try to start a tutorial
  * :warning: Default username is empty, starting the game doesn't work
* Set the username in settings
  * :warning: Had to restart for this to take effect
* Start a game
* :warning: RoundSquare is not detected (I see it installed in `~/snap/freeciv21/x1/.local/share/freeciv21/3.0/`)
* Connect to a LAN game
* Connect to LT76
* Moved the chat around
* Restart the client, start new game
* Chat position is remembered correctly

I built a 55MB snap that seems to work fine, I added comments inline.

File: dist/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+#   - cd <source_dir>/build
+#   - snapcraft
+#   - sudo snap install --devmode ./freeciv21_*.snap
+#
+##########
+
+name: freeciv21
+summary: Freeciv21 - Freeciv for the 21st Century
+description: |
+  Freeciv21 is a free and open source empire-building strategy game inspired by the history of human civilization. The game commences in prehistory and your mission is to lead your tribe from the Stone Age to the Space Age.
```
[the remark this answers is not in the document]
Comment (jwrober) [reply dist/snapcraft.yaml:22]: I tailored the description for the snap instead of copying the metadata from the client xml file. Felt that was better.

**Comment by jwrober** [comment]
On my local, I have the changes suggested. I do not have the modpack icon problem. I do have the user name problem, I think that is an issue in how our options rc file gets created. Same happens in a native install too. I think the freeciv_data_path is causing problems. Let me see if I can get rid of the env lines.

**Comment by jwrober** [comment]
I think I will close this PR and open a new one in a branch from my local.

**Pull request closed without merging**